core/src/services/routing.js

```diff
@@ -299,6 +299,44 @@ export const handleRouteChange = async (path, component, node, config) => {
     return;
   }
   try {
+    if (
+      component &&
+      typeof component.get === 'function' &&
+      component.get().isDirty
+    ) {
+      const newUrl = window.location.href;
+      const oldUrl = component.get().persistUrl;
+
+      //pretend the url hasn't been changed
```

Comment:
There is one scenario where it does not work well. When you keep pressing the browser back button and "No" in the dialog you land on the login page after a while.

Reply:
I'm working on it...